### PR TITLE
[ML][Data Frame] fix progress measurement for continuous transforms

### PR DESCRIPTION
**Pull request by benwtrent**
Progress is now reset on every new checkpoint. 

Some refactoring was necessary so that the changed buckets could be determined **BEFORE** we reset the progress statistics.

### CODE REVIEW

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java [hunk 1]
```diff
@@ -152,7 +130,7 @@ protected boolean initialRun() {
     @Override
     protected void onFinish(ActionListener<Void> listener) {
         // reset the page size, so we do not memorize a low page size forever, the pagesize will be re-calculated on start
-        pageSize = 0;
+        pageSize = pivot.getInitialPageSize();
```
Comment (hendrikmuhs) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java]: I find it more logical to re-calc it at start (line 118). It doesn't matter right now but maybe in future the page size calculation depends on a setting or entails some more other data dependent calculation.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/1

**Comment by benwtrent** [comment]
run elasticsearch-ci/1

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java (hunk 1 above)
Comment (hendrikmuhs) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java]: so the comment "the pagesize will be re-calculated on start" is not true anymore after this change

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/1

**Comment by benwtrent** [comment]
elasticsearch-ci/1

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample

**Pull request merged**